services/auth/source/oauth2/providers.go

```diff
@@ -83,13 +89,43 @@ func RegisterGothProvider(provider GothProvider) {
 	gothProviders[provider.Name()] = provider
 }
 
+// hasExistingAzureADAuthSources checks if there are any existing Azure AD auth sources configured
+func hasExistingAzureADAuthSources(ctx context.Context) bool {
```

Comment:
Maybe we should return the existed types as `[]string` but not only the `bool` and then just list the exists providers but all the 3 types.

Reply:
I chose the structure I did because then we don’t need to loop over slice for each provider instead we can have o(1) of checking if the value from map is true.

Reply:
Ok, updated